### PR TITLE
Nested classes bug fix

### PR DESCRIPTION
**Pull request by RduMarais**

### CODE REVIEW

**Comment by RduMarais** [comment]
test

**Comment by virtuald** [comment]
Thanks for the fix!

Please add a testcase to https://github.com/robotpy/robotpy-cppheaderparser/blob/master/test/test_CppHeaderParser.py so we don't accidentally break it in the future. The easiest way to do this is to copy/paste one of the existing testcases (towards the bottom) and then modify it to fit your usage.

File: CppHeaderParser/CppHeaderParser.py [hunk 1]
```diff
@@ -2453,15 +2453,15 @@ def _evaluate_class_stack(self):
             del self.nameStack[0]
 
         if len(self.nameStack) == 1:
-            if self.nameStack[0] == "struct":
+            if "struct" in self.nameStack:
```
Comment (virtuald) [inline CppHeaderParser/CppHeaderParser.py]: This doesn't make sense. If len(self.nameStack) is 1, then there should only be a single element?

Comment (RduMarais) [reply CppHeaderParser/CppHeaderParser.py]: This was in anticipation for fixing the "parser does not parse nested classes with preprocessor keywords" bug, which needs more work to get fixed.
However, this is more general and is therefore a better practice imo ?

Comment (virtuald) [reply CppHeaderParser/CppHeaderParser.py]: I know that CppHeaderParser does a lot of `'name' in namestack`, but I'm trying to get away from that. I'd rather something parse out a separable piece, then check *those* elements instead.

Anyways, this doesn't add any value as is and will just confuse future readers until someone adds the feature you're talking about, so let's remove it for now.

**Comment by virtuald** [comment]
cppheaderparser will be abandoned soon in favor of https://github.com/robotpy/cxxheaderparser (which doesn't have this bug), so I'm closing this. I encourage you to check out the new library, it should be a lot more robust.

**Pull request closed without merging**